addons/mail/static/src/new/core/channel_member_model.js

```diff
@@ -5,10 +5,25 @@
  * @typedef Data
  * @property {number} id
  * @property {number} partnerId
+ * @property {import("@mail/new/core/thread_model").Thread} memberThread
  */
 export class ChannelMember {
-    constructor(_state, { id, partnerId }) {
-        Object.assign(this, { _state, id, partnerId });
+    static insert(state, data) {
+        let channelMember = state.channelMembers[data.id];
+        if (!channelMember) {
+            channelMember = new ChannelMember();
+            state.channelMembers[data.id] = channelMember;
```

Comment:
We need to return the reactive version (which we get from state)
```suggestion
            state.channelMembers[data.id] = new ChannelMember();
            channelMember = state.channelMembers[data.id];
```